sympy/solvers/ode.py

```diff
@@ -5582,6 +5608,40 @@ def _solve_variation_of_parameters(eq, func, order, match):
         psol = trigsimp(psol, deep=True)
     return Eq(f(x), gsol.rhs + psol)
 
+def ode_factorable(eq, func, order, match):
+    r"""
+    Solves equations having a solvable factor.
+
+    This function is used to solve the equation having factors. Factors may be of type algebraic or ode. It
+    will try to solve each factor independently. Factors will be solved by calling dsolve. We will return the
+    list of solutions.
+
+    Examples
+    ========
+
+    >>> from sympy import Function, dsolve, Eq, pprint, Derivative
+    >>> from sympy.abc import x
+    >>> f = Function('f')
+    >>> eq = (f(x)**2-4)*(f(x).diff(x)+f(x))
+    >>> pprint(dsolve(eq, f(x)))
+                                     -x
+    [f(x) = 2, f(x) = -2, f(x) = C1*e  ]
+
+
+    """
+    eqns = match['eqns']
+    sols = []
+    for eq in eqns:
+        try:
+            sol =dsolve(eq, func)
+            if isinstance(sol, list):
+                sols.extand(sol)
```

Comment:
I guess you don't have any tests covering this branch: extand -> extend

A test could be `dsolve((f(x).diff(x)**2-1)*(f(x).diff(x)**2-4))`

You should also have a test for the NotImplementedError below which could be: `dsolve((f(x).diff(x, 2)-exp(f(x)))*f(x).diff(x))`